smartlearner/utils.py

```diff
@@ -17,14 +18,41 @@ def sharedX(value, name=None, borrow=True, keep_on_cpu=False):
                          borrow=borrow)
 
 
+class NumpyEncoder(json.JSONEncoder):
+    def default(self, obj):
+        if isinstance(obj, np.ndarray):
+            return {"__ndarray__": obj.tolist()}
+
+        return json.JSONEncoder(self, obj)
+
+
+def json_numpy_obj_hook(dct):
+    if isinstance(dct, dict) and '__ndarray__' in dct:
+        return np.array(dct['__ndarray__'])
+
+    return dct
+
+
 def save_dict_to_json_file(path, dictionary):
```

Comment:
Those are implicitly used when calling `save_dict_to_json_file` and `load_dict_from_json_file` with a dictionary containing any numpy array.